content/02-new-to-cardano/10-what-is-a-smart-contract.mdx

```diff
@@ -33,15 +23,16 @@ contracts using such programming languages as:
   financial contracts that allows building contracts visually as well as in more
   traditional code. Financial institutions can use it to develop and deploy
   custom instruments for their customers and clients, for example. The Marlowe
-  language itself is now embedded in both JavaScript and Haskell offering a
+  language itself is now embedded in both JavaScript, TypeScript and Haskell offering a
   choice of editors depending on developers’ preference and skillset.
-  
-Currently, there are more smart contract languages and tools to build on Cardano. For more details, see:
+- **Aiken** - a language and toolchain favoring developer experience. Aiken is used for on-chain validator scripts only.
+- **OpShin** - a programming language for generic smart contracts based on Python. 
```

Comment:
- **[OpShin](https://github.com/OpShin)** - a programming language based on Python for generic smart contracts for Cardano.